File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = [] 
        strMap = {}
        for s in strs: 
            sortedS = ''.join(sorted(s))
            if sortedS in strMap:
                strMap[sortedS].append(s)
            else: 
                strMap[sortedS] = []
                strMap[sortedS].append(s) 
        for key in strMap: 
            ans.append(strMap[key])

        return ans

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int pr= 1; 
        int [] ans = new int [nums.length];
        int flag = 0;  
        for(int i=0; i<nums.length; i++){
            if(nums[i] == 0) flag++; 
            else pr *=nums[i]; 
        }
        for(int i=0; i<nums.length; i++){
            if( nums[i] == 0 && flag <2) ans[i] = pr; 
            else if ( flag >= 1  ) nums[i] = 0; 
            else ans[i] = pr / nums[i]; 
        }
        return ans; 
    }
}  
